app/models/history.rb

```diff
@@ -107,6 +107,169 @@ def self.record_automatically_closed(patient: nil, created_by: 'Sara Alert Syste
+  end
+
+  def self.jurisdiction(history)
+    field = {
+      name: 'Jurisdiction',
+      old_value: Jurisdiction.find(history[:patient][:jurisdiction_id])[:path],
+      new_value: Jurisdiction.find(history[:params][:jurisdiction])[:path]
+    }
+    return if field[:old_value] == field[:new_value]
+
+    create_history(history[:patient], history[:created_by], HISTORY_TYPES[:monitoring_change], compose_message(history, field))
+  end
+
+  def self.assigned_user(history)
+    field = {
+      name: 'Assigned User',
+      old_value: history[:patient][:assigned_user],
+      new_value: history[:params][:assigned_user]
+    }
+    return if field[:old_value] == field[:new_value]
+
+    create_history(history[:patient], history[:created_by], HISTORY_TYPES[:monitoring_change], compose_message(history, field))
+  end
+
+  def self.pause_notifications(history)
+    field = {
+      name: 'notification status',
+      old_value: history[:patient][:pause_notifications] ? 'paused' : 'resumed',
+      new_value: history[:params][:pause_notifications] ? 'paused' : 'resumed'
+    }
+    return if field[:old_value] == field[:new_value]
+
+    create_history(history[:patient], history[:created_by], HISTORY_TYPES[:monitoring_change], compose_message(history, field))
+  end
+
+  def self.symptom_onset(history)
+    field = {
+      name: 'Symptom Onset Date',
+      type: 'date',
+      old_value: history[:patient][:symptom_onset]&.to_date&.strftime('%m/%d/%Y'),
+      new_value: history[:params][:symptom_onset]&.to_date&.strftime('%m/%d/%Y')
+    }
+    return if field[:old_value] == field[:new_value]
+
+    comment = compose_message(history, field)
+    comment += ' The system will now populate this date.' if field[:new_value].nil?
+    create_history(history[:patient], history[:created_by], HISTORY_TYPES[:monitoring_change], comment)
+  end
+
+  def self.last_date_of_exposure(history)
+    field = {
+      name: 'Last Date of Exposure',
+      type: 'date',
+      old_value: history[:patient][:last_date_of_exposure]&.to_date&.strftime('%m/%d/%Y'),
+      new_value: history[:params][:last_date_of_exposure]&.to_date&.strftime('%m/%d/%Y')
+    }
+    return if field[:old_value] == field[:new_value]
+
+    create_history(history[:patient], history[:created_by], HISTORY_TYPES[:monitoring_change], compose_message(history, field))
+  end
+
+  def self.continuous_exposure(history)
+    field = {
+      name: 'Continuous Exposure',
+      old_value: history[:patient][:continuous_exposure] ? 'on' : 'off',
+      new_value: history[:params][:continuous_exposure] ? 'on' : 'off'
+    }
+    return if field[:old_value] == field[:new_value]
+
+    creator = history[:household] == :patient ? 'User' : 'System'
+    formatted_new_value = field[:new_value] ? 'on' : 'off'
+    comment = "#{creator} turned #{formatted_new_value} #{field}#{compose_explanation(history, field)}"
+    create_history(patient, created_by, HISTORY_TYPES[:monitoring_change], comment)
```

Comment:
This currently throws an error. Should be `history[:patient]`.

Reply:
thanks, fixed!